lib/ohm/jamscript/activities.js

```diff
@@ -276,8 +276,12 @@ module.exports = {
             const returnTypeJamLibCode = types.getJamlibCode(returnType);
             cOut += `${returnType} ${functionName}(${functionParams.map((v) => `${v.type} ${v.name}`).join(',')}){`;
             cOut += `arg_t* a=remote_sync_call(cnode->tboard,"${functionName}", "${functionSignature}", ${functionParams.map(v => v.name).join(',')});`;
-            cOut += `${returnType} rval=a->val.${returnTypeJamLibCode};`;
-            cOut += `free(a);`;
+			if (returnType === 'char*') {
+				cOut += `${returnType} rval=strdup(a->val.${returnTypeJamLibCode});`; // TODO try to fix this memory leak, or responsibility of programmer?
```

Comment:
realistically doing this automatically is pretty impossible i think we just indicate in language spec that string results from remote function calls need to be freed?